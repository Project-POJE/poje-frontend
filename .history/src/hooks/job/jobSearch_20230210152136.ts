export function useJobSearchPrefetch(keyword: string) {
  //Todo 검색결과 프리패치 함수
}

export function useJobSearch(keyword: string) {
  //Todo 검색결과 가져오는 함수
}
