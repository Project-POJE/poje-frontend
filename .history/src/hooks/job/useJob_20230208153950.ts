export default function useJob() {
  return <div></div>;
}
