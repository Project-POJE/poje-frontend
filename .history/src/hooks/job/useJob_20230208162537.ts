import { useQuery } from '@tanstack/react-query';
interface UseJobType {
  job: string;
}
export default function useJob({ job }: UseJobType) {
  const {} = useQuery();
  return 123;
}
