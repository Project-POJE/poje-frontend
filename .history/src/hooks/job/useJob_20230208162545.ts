import { useQuery } from '@tanstack/react-query';
interface UseJobType {
  job: string;
}
export default function useJob({ job }: UseJobType) {
  //const { data } = useQuery();
  return 123;
}
