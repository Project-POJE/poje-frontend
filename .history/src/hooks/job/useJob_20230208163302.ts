import service from '@src/service';
import { useQuery } from '@tanstack/react-query';
interface UseJobType {
  job: string;
}
export default function useJob({ job }: UseJobType) {
  const { data } = useQuery(['catefory'], async () => {
    return service.job.getJobCategory();
	}, {
		suspense: true;
	});
  return 123;
}
