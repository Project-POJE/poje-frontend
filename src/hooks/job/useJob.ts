export default function useJob() {
  return 123;
}
