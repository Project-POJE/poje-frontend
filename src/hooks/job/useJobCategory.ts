import service from '@src/service';
import { useQuery } from '@tanstack/react-query';

export default function useJobCategory() {
  const {
    data: category = [],
    isLoading,
    isSuccess,
  } = useQuery(
    ['category'],
    () => {
      return service.job.getJobCategory();
    },
    {
      onError: (e: unknown) => {
        console.log('에러에요에어에어에어에어에어에어에어');
      },
    }
  );
  return { category, isLoading, isSuccess };
}
