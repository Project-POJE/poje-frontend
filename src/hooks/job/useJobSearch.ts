import { useCallback } from 'react';
export default function useJobSearch() {
  const prefetchJobSearch = useCallback((keyword: string) => {
    //Todo 검색결과 프리패치 함수
    console.log('프리패칭 수행중 ...');
  }, []);

  const getJobListByKeyword = useCallback((keyword: string) => {
    //Todo 검색결과 반환하기
  }, []);
  return { prefetchJobSearch, getJobListByKeyword };
}
