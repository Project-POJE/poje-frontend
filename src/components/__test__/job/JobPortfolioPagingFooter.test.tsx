import JobPortfolioPagingFooter from '@src/components/home/JobPage/JobPortfolioPagingFooter';
import { mockPageging } from '@src/test_util/mock/data/job';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useNavigate } from 'react-router-dom';

jest.mock('react-router-dom');
const mockUseNavigate: jest.Mock =
  useNavigate as unknown as jest.MockedFunction<any>;

describe('JobPortfolioPagingFooter Component', () => {
  afterEach(() => {
    mockUseNavigate.mockReset();
  });

  it('1~5 페이지 출력', () => {
    const { asFragment } = render(
      <JobPortfolioPagingFooter {...mockPageging} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('페이지 클릭시 해당 페이지 번호 페이지로 이동', () => {
    const mockNavigate = jest.fn();
    mockUseNavigate.mockImplementation(() => mockNavigate);

    render(<JobPortfolioPagingFooter {...mockPageging} />);

    // 다음 버튼을 클릭하면 moveNextPage 함수가 호출되는지 확인
    const nextButton = screen.getByTestId('next-btn');
    userEvent.click(nextButton);
    expect(mockNavigate).toHaveBeenCalledWith(`${mockPageging.path}/6`);

    // 마지막 페이지로 이동 버튼을 클릭하면 moveEndPage 함수가 호출되는지 확인
    const nextEndButton = screen.getByTestId('next-end-btn');
    userEvent.click(nextEndButton);
    expect(mockNavigate).toHaveBeenCalledWith(
      `${mockPageging.path}/${mockPageging.totalPageCnt}`
    );
  });
});
