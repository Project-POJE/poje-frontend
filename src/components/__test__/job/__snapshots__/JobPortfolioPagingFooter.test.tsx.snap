// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobPortfolioPagingFooter Component 1~5 페이지 출력 1`] = `
<DocumentFragment>
  <ul
    class="sc-bcXHqe kdpBQo"
  >
    <li
      class="sc-gswNZR bMITra"
      data-testid="prev-end-btn"
    >
      &lt;&lt;
    </li>
    <li
      class="sc-dkrFOg kQgduz"
    >
      1
    </li>
    <li
      class="sc-dkrFOg cmIpsB"
    >
      2
    </li>
    <li
      class="sc-dkrFOg cmIpsB"
    >
      3
    </li>
    <li
      class="sc-dkrFOg cmIpsB"
    >
      4
    </li>
    <li
      class="sc-dkrFOg cmIpsB"
    >
      5
    </li>
    <li
      class="sc-gswNZR bMITra"
      data-testid="next-btn"
    >
      &gt;
    </li>
    <li
      class="sc-gswNZR bMITra"
      data-testid="next-end-btn"
    >
      &gt;&gt;
    </li>
  </ul>
</DocumentFragment>
`;
